--- a/modules/arr-common/hostConfig.nix
+++ b/modules/arr-common/hostConfig.nix
@@ -326,6 +326,7 @@
             NEW_CONFIG=$(${pkgs.jq}/bin/jq -n \
               ${jqSecrets.flagsString} \
               --argjson id "$CONFIG_ID" \
+              --arg allowedHosts "$(echo "$HOST_CONFIG" | ${pkgs.jq}/bin/jq -r '.allowedHosts // ""')" \
               --arg bindAddress ${escapeShellArg hc.bindAddress} \
               --arg authenticationMethod ${escapeShellArg hc.authenticationMethod} \
               --arg authenticationRequired ${escapeShellArg hc.authenticationRequired} \
@@ -345,6 +346,7 @@
               --arg backupFolder ${escapeShellArg hc.backupFolder} \
               '{
                 id: $id,
+                allowedHosts: $allowedHosts,
                 bindAddress: $bindAddress,
                 port: ${builtins.toString hc.port},
                 sslPort: ${builtins.toString hc.sslPort},
